src/braket/emulators/emulator_passes/criteria/connectivity_criterion.py

```diff
@@ -0,0 +1,143 @@
+from braket.emulators.emulator_passes.criteria.emulator_criterion import EmulatorCriterion
+from braket.registers.qubit_set import QubitSet
+
+
+class ConnectivityCriterion(EmulatorCriterion):
+    def __init__(
+        self,
+        connectivity_graph: Union[Dict[int, Iterable[int]], DiGraph] = None,
+        fully_connected=False,
+        num_qubits: int = None,
+        qubit_labels: Union[Iterable[int], QubitSet] = None,
+        directed: bool = True,
+    ):
+        """
+        args:
+            connectivity_graph (Union[Dict[int, List[int]], DiGraph]): Either a sparse matrix or
+            DiGraph representation of the device connectivity. Can be None if fully_connected is
+            true.
+
+            fully_connected (bool): If true, the all qubits in the device are connected.
+
+            num_qubits (int): The number of qubits in the device; if fully_connected is True,
+            this is used to create a complete graph with num_qubits nodes; ignored if
+            connectivity_graph is provided and fully_connected if False.
+
+            qubit_labels (Iterable[int]): A set of qubit labels; if fully_connected is True,
+            the qubits_labels are used as nodes of a fully connected topology; ignored if
+            connectivity_graph is provided and fully_connected if False.
+
+            directed (bool): Denotes if the connectivity graph is directed or undirected. If
+                the connectivity graph is undirected, this constructor attempts to fill in any
+                missing back edges.
+        """
+
+        if not (connectivity_graph or fully_connected):
+            raise ValueError(
+                "Either the connectivity_graph must be provided or fully_connected must be True."
+            )
+
+        if fully_connected:
+            if not ((num_qubits is None) ^ (qubit_labels is None)):
+                raise ValueError(
+                    "Either num_qubits or qubit_labels (NOT both) must be \
+                        provided if fully_connected is True."
+                )
+            self._connectivity_graph = complete_graph(
+                num_qubits if num_qubits else qubit_labels, create_using=DiGraph()
+            )
+        elif not isinstance(connectivity_graph, DiGraph):
+            try:
+                self._connectivity_graph = from_dict_of_lists(
+                    connectivity_graph, create_using=DiGraph()
+                )
+            except Exception as e:
+                raise ValueError(
+                    f"connectivity_graph must be a valid DiGraph or a dictionary\
+                        mapping integers (nodes) to a list of integers (adjancency lists): {e}"
+                )
+        else:
+            self._connectivity_graph = connectivity_graph
+
+        if not directed:
+            for edge in self._connectivity_graph.edges:
+                self._connectivity_graph.add_edge(edge[1], edge[0])
+
+    def validate(self, circuit: Circuit) -> None:
```

Comment:
Should this also have an abstract parent method?

Reply:
Are you referring to `validate()` for the `ConnectivityCriterion` class? If so, `ConnectivityCriterion.validate()` implements the abstractmethod, `EmulatorCriterion.validate` from the `EmulatorCriterion` class that `ConnectivityCriterion` inherits from. 

I'm not sure if this answers your question?